### ISSUE TITLE
Client (renderer) Freeze on Dimension Packet when using Linux

### ISSUE THREAD

**Issue by 0x277F** [MinecraftForge/MinecraftForge#3157]
This is a very specific issue that I've had a lot of trouble reproducing. Some information can be found at https://bugs.mojang.com/browse/MC-101266 (which was closed because the mojira mods thought it was an issue with a modded server), but I'll try and give as much backstory as possible. Around the 1.8 update I began experiencing freezes while changing servers on a BungeeCord network. Because I don't play Minecraft very much I didn't give it much thought, but more recently I saw the issue posted on the Mojang issue tracker which pretty much exactly summed up the issue I was having. Specifically - when the client receives and processes the _second_ respawn packet during its lifetime. I've asked several people to try and reproduce this and they've been unable to unless they're running Linux. It is present on all versions after 1.8 as far as I can tell (the following was produced on 1.9.4).

The original issue on Mojira has a profiler output which is relevant. I've managed to get a small amount of profiling done (but was unable to continue due to a really bad computer), along with a fair bit of sampling. The profiled one has some mods (including LiteLoader ones), but the sampled one does not and results are identical. Some links:
- A VisualVM profiler record: https://assets.hex.lc/perm/respawn-packet-one.nps
- A VisualVM sampler record: https://assets.hex.lc/perm/respawn-packet-two-sampled.nps
- A screenshot showing what (I think) the issue is from the former: http://i.imgur.com/uPeXy9s.png
- A similar screenshot from the latter: http://i.imgur.com/v3uCZ0J.png
- An alternative issue from the latter: http://i.imgur.com/66Djawk.png
- Same scenario as the previous screenshot on vanilla: http://i.imgur.com/gEAuw3E.png

I saw the ChunkRenderDispatcher methods initially, but some further thought leads me to believe their fault is unlikely. I share them simply because they may be of use later. The final screenshot I posted is pretty much the same issue as the one posted on Mojira, so I believe this is the problem. This is especially possible in my mind because the issue is only present on Linux - meaning a native method is more likely the culprit.

As you'll see, however, the issue isn't anywhere close to being problematic on a Vanilla client. This leads me to my question - is there something Forge does that would cause this? I don't see any outstanding patches to Minecraft client code that would be likely to, but I'm not experienced with client modding to that extent. 

TL;DR - When the client respawns on Forge, http://i.imgur.com/66Djawk.png compared to http://i.imgur.com/gEAuw3E.png on a Vanilla client.

Edit: Relevant post I found: https://bugs.mojang.com/browse/MC-31714

**Comment by 0x277F** [comment]
According to #mcdevs, 
```
<+SinZ> Oct 23 18:36:44 <Grum> -        int maxChunksWidth = (MAX_VIEW_DISTANCE + 1) * 2; // Why +1?
<+SinZ> Oct 23 18:36:44 <Grum> +        int maxChunksWidth = (Options.RENDER_DISTANCE_EXTREME + 1) * 2; // Why +1?
```
is the patch that fixed the bug I linked in my edit. Apparently the previous issue was caused by attempted overallocation of GL lists, which Windows drivers apparently ignored (or handled better than linux drivers).

**Comment by 0x277F** [comment]
I've done some more profiling (and made sure Optifine is disabled). All profiling was done with a render distance of 12, fast graphics, a 60fps frame limit, and smooth lighting disabled. All other settings are default.

- Forge client, vanilla server: https://assets.hex.lc/perm/respawn-no-of-vanilla-s.nps
- Screenshot of above: http://i.imgur.com/SkOzWmX.png
- Vanilla client, vanilla server: https://assets.hex.lc/perm/respawn-vanilla-v-s.nps 
- Screenshot of above: http://i.imgur.com/G7B2BVQ.png
- Forge client, BungeeCord server: https://assets.hex.lc/perm/respawn-no-of-bungee-s.nps
- Screenshot of above: http://i.imgur.com/rbAWoZU.png
- Vanilla client, BungeeCord server: https://assets.hex.lc/perm/respawn-vanilla-bungee-s.nps
- Screenshot of above: http://i.imgur.com/wGpBf3I.png

Results were a bit unpredictable, I'm going to do some analysis and post what I find.

**Comment by Actuarius** [comment]
@williewillus added labels [Bug]
